### PR TITLE
Int2019 57 request funds reciever view

### PR DESCRIPTION
**Pull request by radulambrino**

### CODE REVIEW

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 1]
```diff
@@ -189,17 +249,20 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
             })
             .then(response => response.json())
             .then( response => {
+              console.log("continued the shit");
```
Comment (Slitthe) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: or maybe not....

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 2]
```diff
@@ -8,10 +8,10 @@ import '../ExchangeForm.css';
 import "../../Common/styles.css";
 import TransferFundsModal from './TransferFundsModal';
 import Toggle from '../../Common/Toggle';
+import TransferRequests from '../TransferRequests';
 
-const transferFundsUrl = `${constants.baseUrl}transferFunds`;
-const transferRequestsUrl = `${constants.baseUrl}transferRequests`;
-const bankAccountUrl = `${constants.baseUrl}bankAccount`;
+const transferUrl = `${constants.baseUrl}transferFunds`;
+const requestUrl = `${constants.baseUrl}transferRequests`;
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: I think transferRequestsUrl and transferFundsUrl were better names

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 3]
```diff
@@ -105,7 +122,7 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
     getTotalBalance = () => {
       let token = sessionStorage.getItem('Authentication-Token');
         if(token != null) {
-            fetch(bankAccountUrl, {
+            fetch(constants.baseUrl+"bankAccount", {
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: can't you use bankAccountUrl?

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 4]
```diff
@@ -122,6 +139,25 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
...
+        cardNumber: transferData.cardNumberRequester,
+        transferDetails: "Responding to your request to transfer money.",
+        requestTransfer: false,
+        isTransferRequested: true,
+        requestId: transferData.id
+      })
+    }
+
+    changeReloading = (reload: boolean) =>
+    {
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: move open brace on same line with  changeReloading = (reload: boolean) =>

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 5]
```diff
@@ -144,42 +180,66 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
...
-              .then( response => {            
-                  this.setState({
-                    openModal: true,
-                    modalMessage: response
-                  }); 
-                  
-                  this.handleIsButtonDisabled(); 
-              });
-          }
+      let data = this.getDataForTransfer();
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: declare with const, it's a readonly variable

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 6]
```diff
@@ -144,42 +180,66 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
...
+
+      if(this.isValidAmount(data)) {
+
+        let token = sessionStorage.getItem('Authentication-Token');
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: declare with const, it's a readonly variable

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 7]
```diff
@@ -144,42 +180,66 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
...
+            .then(response => response.json())
+            .then( response => {
+              
+                this.setState({
+                  openModal: true,
+                  modalMessage: response
+                }); 
+                
+                if(this.state.isTransferRequested)
+                {
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: move open brace on same line with if

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 8]
```diff
@@ -144,42 +180,66 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
...
+                this.handleIsButtonDisabled(); 
+            });
         }
+      }
     }
 
-    requestTransfer = () => {
+    requestTransfer = () =>
+    {
       let data = this.getTransferDataFromUI();
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: declare with const, it's a readonly variable

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 9]
```diff
@@ -144,42 +180,66 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
...
+
       if(this.isValidAmount(data)) {
+
         let token = sessionStorage.getItem('Authentication-Token');
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: declare with const, it's a readonly variable

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 10]
```diff
@@ -144,42 +180,66 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
...
+      }
     }
 
-    requestTransfer = () => {
+    requestTransfer = () =>
+    {
       let data = this.getTransferDataFromUI();
+
       if(this.isValidAmount(data)) {
+
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: no empty line needed

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 11]
```diff
@@ -144,42 +180,66 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
...
         let token = sessionStorage.getItem('Authentication-Token');
+
          if(token !== null)
         {
           this.handleIsButtonDisabled();
-          fetch(transferRequestsUrl, {
+          console.log("Started the shit");
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: remove console.log

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 12]
```diff
@@ -190,17 +250,20 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
...
+
                 this.handleIsButtonDisabled();
             });
         } 
       }
     }
 
-    closeModal = () => {
+    closeModal = () =>
+    {
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: closeModal = () => { it's fine in javascript. Change it back :)

File: ClientApp/viking-vault/src/Login.tsx [hunk 13]
```diff
@@ -126,8 +126,7 @@ class LoginForm extends React.Component<any, ILoginFormState> {
     })
     .then(result => {
       if(result !== null)
-      {
-        // correct email + password => redirect to user/admin page 
+      {
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/Login.tsx]: move open brace on same line with if

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 14]
```diff
@@ -63,10 +69,21 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
...
+
+        if(this.state.requestTransfer) {
+            if(Number (e.target.value) >= 0) {
+                this.setState({
+                  transferedAmount: e.target.value
+                });
+            }            
+        }
+        else
+        {
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: move brace on same line with else

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 15]
```diff
@@ -144,38 +179,58 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
...
+            headers: {
+              'Accept': 'application/json',
+              'Content-Type': 'application/json',
+              'x-access-token': token.toString()
+            },
+            body: JSON.stringify(data)
+            })
+            .then(response => response.json())
+            .then( response => {
+
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: remove empty line

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 16]
```diff
@@ -144,38 +179,58 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
...
+      const data = this.getDataForTransfer();
+
+      if(this.isValidAmount(data)) {
+
+        const token = sessionStorage.getItem('Authentication-Token');
+
+        if(token !== null) {
+
+          this.handleIsButtonDisabled();
+
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: remove empty line

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 17]
```diff
@@ -144,38 +179,58 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
...
-              });
-          }
+      const data = this.getDataForTransfer();
+
+      if(this.isValidAmount(data)) {
+
+        const token = sessionStorage.getItem('Authentication-Token');
+
+        if(token !== null) {
+
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: remve empty line

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 18]
```diff
@@ -144,38 +179,58 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
...
-                    modalMessage: response
-                  }); 
-                  
-                  this.handleIsButtonDisabled(); 
-              });
-          }
+      const data = this.getDataForTransfer();
+
+      if(this.isValidAmount(data)) {
+
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: remove empty line

File: ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx [hunk 19]
```diff
@@ -209,30 +265,34 @@ class TransferFundsPage extends React.Component<any, ITransferFundsState>{
     }
 
     render(){
+
```
Comment (crisfsabau) [inline ClientApp/viking-vault/src/UserDashboard/TransferFunds/TransferFundsPage.tsx]: remove empty line

**Pull request merged**